### ISSUE TITLE
 Validation error with TypeAdapter of TypedDict

### ISSUE THREAD

**Issue by olivier65** [pydantic/pydantic#8946]
### Initial Checks

- [X] I confirm that I'm using Pydantic V2

### Description

Pydantic doesn't handle the Sequence type properly. The code below yields

```
pydantic_core._pydantic_core.ValidationError: 1 validation error for typed-dict
x
  Input should be a valid list [type=list_type, input_value=('abc',), input_type=tuple]
    For further information visit https://errors.pydantic.dev/2.6/v/list_type
```

yet tuples are of course sequences.

### Example Code

```Python
from typing import Sequence

from pydantic import TypeAdapter
from typing_extensions import TypedDict


class Model(TypedDict):
    x: Sequence[str]


validator = TypeAdapter(Model)
validator.validate_python({'x': ('abc',)}, strict=True)
```

### Python, Pydantic & OS Version

```Text
pydantic version: 2.6.3
        pydantic-core version: 2.16.3
          pydantic-core build: profile=release pgo=true
                 install path: C:\Users\Me\venv\Lib\site-packages\pydantic
               python version: 3.11.7 (tags/v3.11.7:fa7a6f2, Dec  4 2023, 19:24:49) [MSC v.1937 64 bit (AMD64)]
                     platform: Windows-10-10.0.19045-SP0
             related packages: typing_extensions-4.9.0
                       commit: unknown
```

**Comment by sydney-runkle** [comment]
Marking as resolved, this is fixed on `main` and said fix will be released with 2.7!

**Comment by stenon-olivier** [comment]
Is the fix already in 2.6.4 perchance?

**Comment by sydney-runkle** [comment]
@stenon-olivier,

It's not, but should be released with the 2.7 beta hopefully later this week :).
